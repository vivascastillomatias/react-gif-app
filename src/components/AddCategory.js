import React, {useState} from 'react'

import PropTypes from 'prop-types';


export function AddCategory({setCategories}) {

    const [inputValue, setInputValue] = useState('')
    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 2) {
            setCategories((categories) => [inputValue,...categories])
            setInputValue('')
        }
    }
    return (
        <>
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <h5>Add Category</h5>
            <input 
                type='text'
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}