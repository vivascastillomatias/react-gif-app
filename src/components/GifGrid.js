// import React, {useState, useEffect} from 'react'
import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs'
// import getGifs from '../helpers/getGifs';
import GifGridItem from './GifGridItem';

export default function GifGrid({category}) {

    // const [images, setImages] = useState([])

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <h3 className='animate__animated animate__fadeIn'>{category}</h3>
        { loading && <p className='animate__animated animate__flash'>loading...</p>}
        <div className='card-grid'>
            {
                images.map((img) => {
                    return <GifGridItem key={img.id} {...img}></GifGridItem>
                })
            }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
