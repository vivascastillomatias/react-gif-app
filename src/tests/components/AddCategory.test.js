import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Pruebas en <AddCategory />', () => {
    
    const setCategories = jest.fn();
    let wrapper =  shallow(<AddCategory setCategories={setCategories}/>)

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper =  shallow(<AddCategory setCategories={setCategories}/>) 
    })
    
    test('Construcción correcta del componente', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe cambiar el valor de la caja de texto | handleInputChange', () => {

        const input = wrapper.find('input')
        const value = "1111111"
        input.simulate('change', {target: {value}})
        const change = wrapper.find('p').text();

        expect(change).toBe(value)
    })

    test('No debe de postear la informacion en el onSubmit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}})
        expect(setCategories).not.toHaveBeenCalled() // Saber si una funcion fue llamada una vez o más
    })
    
    test('Debe llamar a setCategories y limpiar la caja de texto', () => {
        const value = "Shrek"
        wrapper.find('input').simulate('change', {target: {value}})
        wrapper.find('form').simulate('submit', {preventDefault(){}})
        expect(setCategories).toHaveBeenCalled();
        // expect(setCategories).toHaveBeenCalledTimes(k) // Saber si la funcion fue llamada k veces
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function) ); // Saber si la funcion fue llamada con una funcion
        

        expect(wrapper.find('input').prop('value')).toBe('')
    })
    

})
