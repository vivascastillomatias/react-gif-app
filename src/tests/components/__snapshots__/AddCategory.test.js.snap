// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Construcci√≥n correcta del componente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <h5>
      Add Category
    </h5>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
